packages/labs/src/tag-input/tagInput.tsx

```diff
@@ -167,8 +170,10 @@ export class TagInput extends AbstractComponent<ITagInputProps, ITagInputState>
         const { selectionEnd, value } = event.currentTarget;
         if (event.which === Keys.ENTER && value.length > 0) {
             // enter key on non-empty string invokes onAdd
-            this.setState({ inputValue: "" });
-            Utils.safeInvoke(this.props.onAdd, value);
+            const shouldClearInput = Utils.safeInvoke(this.props.onAdd, value);
+            if (shouldClearInput) {
```

Comment:
That's fair, I mean I guess the docs speaks for itself now, we really were lacking this `return false` capability